modules/gltf/src/lib/extensions/EXT_structural_metadata.ts

```diff
@@ -0,0 +1,730 @@
+
+/**
+ * Decodes properties of enum type from binary sourse.
+ * @param {GLTF_EXT_structural_metadata_Schema} schema - schema object
+ * @param {GLTF_EXT_structural_metadata_ClassProperty} classProperty - class property object
+ * @param {number} numberOfElements - The number of elements in each property array that propertyTableProperty contains. It's a number of rows in the table.
+ * @param {Uint8Array} valuesDataBytes - data taken from values property of the property table property.
+ * @param {TypedArray | null} arrayOffsets - offsets for variable-length arrays. It's null for fixed-length arrays or scalar types.
+ * @returns {string[] | string[][]}
+ */
+function getPropertyDataENUM(
+  schema: GLTF_EXT_structural_metadata_Schema,
+  classProperty: GLTF_EXT_structural_metadata_ClassProperty,
+  numberOfElements: number,
+  valuesDataBytes: Uint8Array,
+  arrayOffsets: TypedArray | null
+): string[] | string[][] {
+  const data: string[] = [];
+
+  const isArray = classProperty.array;
+  const arrayCount = classProperty.count;
+
+  const enumType = classProperty.enumType;
+  // Enum ID as declared in the `enums` dictionary. Required when `type` is `ENUM`.
+  if (!enumType) {
+    throw new Error(
+      'Incorrect data in the EXT_structural_metadata extension: classProperty.enumType is not set for type ENUM'
+    );
+  }
+
+  const enumEntry: GLTF_EXT_structural_metadata_Enum | undefined = schema.enums?.[enumType];
+  if (!enumEntry) {
+    throw new Error(
+      `Incorrect data in the EXT_structural_metadata extension: schema.enums does't contain ${enumType}`
+    );
+  }
+
+  const enumValueType = enumEntry.valueType || 'UINT16';
+  const elementSize = getArrayElementByteSize(classProperty.type, enumValueType);
+  const elementCount = valuesDataBytes.byteLength / elementSize;
+  const valuesData: TypedArray = convertRawBufferToMetadataArray(
+    valuesDataBytes,
+    classProperty.type,
+    enumValueType,
+    elementCount
+  );
+
+  if (isArray) {
+    if (arrayOffsets) {
+      // VARIABLE-length array
+      return handleVariableLengthArrayENUM(
+        valuesData,
+        numberOfElements,
+        arrayOffsets,
+        valuesDataBytes.length,
+        elementSize,
+        enumEntry
+      );
+    }
+    if (arrayCount) {
+      // FIXED-length array
+      return handleFixedLengthArrayENUM(valuesData, numberOfElements, arrayCount, enumEntry);
+    }
+    return [];
+  }
+
+  // Single value (not an array)
+  for (let index = 0; index < numberOfElements; index++) {
+    const enumValue = valuesData[index];
+    const enumObject = getEnumByValue(enumEntry, enumValue);
+    if (enumObject) {
+      data.push(enumObject.name);
+    }
+  }
+
+  return data;
+}
+
+/* eslint max-params: ["error", 6]*/
+function handleVariableLengthArrayENUM(
+  valuesData: TypedArray,
+  numberOfElements: number,
+  arrayOffsets: TypedArray,
+  valuesDataBytesLength: number,
+  elementSize: number,
+  enumEntry: GLTF_EXT_structural_metadata_Enum
+) {
```

Comment:
Union of all params in one object is a good practice.
```suggestion
function handleVariableLengthArrayENUM(
  params: {
    valuesData: TypedArray;
    numberOfElements: number;
    arrayOffsets: TypedArray;
    valuesDataBytesLength: number;
    elementSize: number;
    enumEntry: GLTF_EXT_structural_metadata_Enum
  }
) {
```